### DISCUSSION TITLE
[SKIA] How to use Hardware Acceleration on Target Framework Desktop?

### DISCUSSION THREAD

**Discussion by MichaelJachan** [unoplatform/uno#17700] [Q&A] [answered]
Hello!

I am working on Win11 with Target Framework `Desktop` with `"Uno.Sdk": "5.3.31"`

I have seen your code example (XAML-Markup) SkiaSharpTest. When running SkiaSharpTest and clicking CheckBox Hardware Accelerated, the Message "is not supported" will be shown. As far as I have understood, `SKSwapChainPanel` cares for hardware acceleration, right? I have searched your code base and I guess, `SKSwapChainPanel` is available on other platforms than Windows. 

I have successfully replicated SkiaSharpTest using C# MARKUP, but only the version with painting on a `SKXamlCanvas` (no HW acceleration). 

My question is: How to use Hardware Acceleration on Windows? 

I can include the type `SKSwapChainPanel `by using NuGet `SkiaSharp.Views`, but then I obtain a runtime Exception `System.NotSupportedException' in SkiaSharp.Views.Windows.dll`

Best, Michael

**Comment by jeromelaban** [comment]
At this time, the implementation for `SKSwapChain` panel is indeed not supported on `net8.0-desktop`, and `SKXamlCanvas` uses software rendering, even if it's rendering to a hardware rendered surface.

We're working on changing this in https://github.com/unoplatform/uno/pull/16621 with `SkiaVisual`, where the underlying SkiaSharp context will be useable by the newer version of `SKXamlCanvas` (or `SKSwapChainPanel`).

**Comment by MartinZikmund** [comment]
Hi, part of this work has been merged, but we have a follow up PR in progress, tracking in https://github.com/unoplatform/uno/pull/18181

**Reply by MichaelJachan** [reply]
Hi @MartinZikmund!

I have pulled newest version of SkiaSharpTest (PR #846) and its behavior is unchanged, it means:

- Under Windows Desktop: The application runs, but when checking Checkbox "Hardware acceleration", it still yields "SKSwapChainPanel is not supported on this platform". 
- Under WinAppSDK unpackaged: The Application crashes with the following message: 
  - The program '[24324] SkiaSharpTest.exe' has exited with code 3221226107 (0xc000027b).

Then I was thinking this was fixed with the newest UNO SDK `5.6.0-dev.26` and I updated (from `5.3.108`). The behavior did not change.

**Reply by MartinZikmund** [reply]
@MichaelJachan Yes, that sample is not up to date unfortunately. Try following the docs here: https://github.com/unoplatform/uno/blob/master/doc/articles/controls/SKCanvasElement.md. We will provide an updated sample in the Uno Samples repository

**Reply by ramezgerges** [reply, marked as answer]
Here's the PR for the sample: https://github.com/unoplatform/Uno.Samples/pull/847. I'm not sure if dicussions are automatically closable the way PRs are, but if not, feel free to close this discussion after the PR is merged.
